src/coreclr/vm/jithelpers.cpp

```diff
@@ -5736,6 +5736,65 @@ HCIMPL3(void, JIT_VTableProfile64, Object* obj, CORINFO_METHOD_HANDLE baseMethod
 }
 HCIMPLEND
 
+HCIMPL1(void, JIT_CountProfile32, volatile LONG* pCounter)
```

Comment:
It would be useful to have comments here, if nothing else, to link to the design paper.